### PR TITLE
Alerting: Improve the docs on templating labels and annotations

### PR DESCRIPTION
**Pull request by grobinson-grafana**
**What is this feature?**

The previous version of this page I wrote with the expectation that readers would first learn the templating language and then write their templates. This doesn't seem to have worked out as well as I had expected, and so I've rewritten the documentation to explain the language using relevant and useful examples instead.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/alerting/fundamentals/annotation-label/variables-label-annotation.md [hunk 1]
```diff
@@ -18,153 +18,255 @@ weight: 117
...
+You can use templates to include data from queries and expressions in labels and annotations. For example, you might want to set the severity label for an alert based on the value of the query, or use the instance label from the query in a summary annotation so you know which server is experiencing high CPU usage.
 
-## Go's templating language
+All templates should be written in [text/template](https://pkg.go.dev/text/template). Regardless or whether you are templating a label or an annotation, you should write each template inline inside the label or annotation that you are templating. This unfortunately means you cannot share templates between labels and annotations, and instead you will need to copy templates wherever you want to use them.
```
Comment (armandgrillet) [inline docs/sources/alerting/fundamentals/annotation-label/variables-label-annotation.md:23]:
```suggestion
All templates should be written in [text/template](https://pkg.go.dev/text/template). Regardless of whether you are templating a label or an annotation, you should write each template inline inside the label or annotation that you are templating. This means you cannot share templates between labels and annotations, and instead you will need to copy templates wherever you want to use them.
```

File: docs/sources/alerting/fundamentals/annotation-label/variables-label-annotation.md [hunk 2]
```diff
@@ -18,153 +18,255 @@ weight: 117
...
 
-Templates for labels and annotations are written in Go's templating language, [text/template](https://pkg.go.dev/text/template).
+Each template is evaluated whenever the alert rule is evaluated, and is evaluated for every alert separately. For example, if your alert rule has a templated summary annotation, and the alert rule has 10 firing alerts, then the template will be executed 10 times, once for each alert. You should therefore avoid doing expensive computations in your templates as much as possible as high cardinality alert rules can execute a template hundreds or even thousands of times every time the alert rule is evaluated, depending on the number of time series.
```
Comment (armandgrillet) [inline docs/sources/alerting/fundamentals/annotation-label/variables-label-annotation.md:25]:
```suggestion
Each template is evaluated whenever the alert rule is evaluated, and is evaluated for every alert separately. For example, if your alert rule has a templated summary annotation, and the alert rule has 10 firing alerts, then the template will be executed 10 times, once for each alert. You should therefore avoid doing expensive computations in your templates, especially with high cardinality alert rules, as templates are executed every time the alert rule is evaluated.
```

File: docs/sources/alerting/fundamentals/annotation-label/variables-label-annotation.md [hunk 3]
```diff
@@ -18,153 +18,255 @@ weight: 117
...
-If you were to print the value of the expression with RefID `B` in the summary of the alert:
+To print the value of an instant query as a percentage use the `humanizePercentage` function:
 
 ```
-{{ $labels.service }} has over 5% of responses with 5xx errors: {{ $values.B }}%
+{{ humanizePercentage (index $values "A").Value }}
 ```
 
-The summary will contain just the value:
+This function expects the value to be a decimal number between 0 and 1. If the value is instead a decimal number between 0 and 100 you can either divide it by 100 in your query or using a math expression. If the query is a range query you must first reduce it from a time series to an instant vector with a reduce expression.
```
Comment (armandgrillet) [inline docs/sources/alerting/fundamentals/annotation-label/variables-label-annotation.md:131]:
```suggestion
This function expects the value to be a decimal number between 0 and 1. If the value is instead a decimal number between 0 and 100 you can divide it by 100 in your query or using a math expression. If the query is a range query you must first reduce it from a time series to an instant vector with a reduce expression.
```

Comment (grobinson-grafana) [reply docs/sources/alerting/fundamentals/annotation-label/variables-label-annotation.md:131]: I think either should be kept here no?

Comment (grobinson-grafana) [reply docs/sources/alerting/fundamentals/annotation-label/variables-label-annotation.md:131]: I think either should be kept here no? Without it the sentence doesn't quite read right.

**Review by brendamuir — APPROVED** [review]

**Pull request merged**